src/styles/day.scss

```diff
@@ -0,0 +1,50 @@
+.calender {
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: $width-week;
+
+  @each $day, $size in $days {
+    .calender--start-day--#{$day}
+    &__day:first-child {
+      margin-left: $size;
+    }
+  }
+
+  @for $index from $min-last-days-month to $max-last-days-month {
+    .calender--month-length--#{$index}
+
+    &__day:nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    @include day-block($width);
+    background-color: #eee;
+    border: 1px solid black;
+    box-sizing: border-box;
+    position: relative;
+
+    @for $index from $first-day through $number-of-days-month {
+      &:nth-child(#{$index})::before {
+        position: absolute;
+        content: "#{$index}";
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+        font-weight: bold;
```

Comment:
Don't add styles that don't change between the boxes in the loop. Resulting css file size will be enormous.